rl4j/rl4j-core/src/main/java/org/deeplearning4j/rl4j/learning/Learning.java

```diff
@@ -64,6 +64,9 @@ public static Integer getMaxAction(INDArray vector) {
     public static <O extends Encodable, A, AS extends ActionSpace<A>> INDArray getInput(MDP<O, A, AS> mdp, O obs) {
         INDArray arr = Nd4j.create(obs.toArray());
         int[] shape = mdp.getObservationSpace().getShape();
+
+        // Review: Why is shape.length == 1 a special case?
+        // Is it because HistoryProcessor expects at least a 2D array?
         if (shape.length == 1)
             return arr.reshape(new long[] {1, arr.length()});
```

Comment:
ND4J didn't use to support arrays of 0 or 1 dimensions, so code in RL4J assumes we have at least 2 dimensions. This could be fixed better obviously.